sql/src/main/java/org/apache/druid/sql/calcite/schema/DruidSchema.java

```diff
@@ -402,7 +402,7 @@ private void addSegment(final DruidServerMetadata server, final DataSegment segm
     }
   }
 
-  private void removeSegment(final DataSegment segment)
+  protected void removeSegment(final DataSegment segment)
```

Comment:
Please add `@VisibleForTesting` to newly exposed methods.

Reply:
Yes, forgot to add. Will do.